app/src/userpages/components/KeyField/index.jsx

```diff
@@ -27,187 +28,164 @@ type Props = {
-                </DropdownActions.Item>,
-            ]),
-            <DropdownActions.Item key="copy" onClick={this.onCopy}>
-                <Translate value="userpages.keyField.copy" />
-            </DropdownActions.Item>,
-            ...useIf(!!allowEdit, [
-                <DropdownActions.Item key="edit" onClick={this.onEdit}>
-                    <Translate value="userpages.keyField.edit" />
-                </DropdownActions.Item>,
-            ]),
-            ...useIf(!!allowDelete, [
-                <DropdownActions.Item key="delete" onClick={this.onDelete} disabled={disableDelete}>
-                    <Translate value="userpages.keyField.delete" />
-                </DropdownActions.Item>,
-            ]),
-        ]
-
-        return (
-            <div
-                className={cx(styles.keyFieldContainer, keyFieldClassName)}
-            >
-                <TextInput
-                    label={keyName}
-                    actions={actions}
-                    value={value && (!truncateValue ? value : truncate(value, {
-                        maxLength: 15,
-                    }))}
-                    readOnly
-                    type={hidden ? 'password' : 'text'}
-                    preserveLabelSpace
-                />
-            </div>
-        )
-    }
-
-    render = () => {
-        const { keyName, value, className, valueLabel } = this.props
-        const { waiting, editing, error } = this.state
-
-        return (
-            <div className={cx(styles.root, styles.KeyField, className)}>
-                {!editing ? (
-                    this.renderInput()
-                ) : (
-                    <KeyFieldEditor
-                        keyName={keyName}
-                        value={value}
-                        onCancel={this.onCancel}
-                        onSave={this.onSave}
-                        waiting={waiting}
-                        error={error}
-                        valueLabel={valueLabel}
+    }, [allowDelete, onDeleteProp])
+
+    const onEdit = useCallback(() => {
+        setEditing(true)
+    }, [])
+
+    const revealAction = useMemo(() => (
+        <DropdownActions.Item key="reveal" onClick={toggleHidden}>
+            <Translate value={`userpages.keyField.${hidden ? 'reveal' : 'conceal'}`} />
+        </DropdownActions.Item>
+    ), [toggleHidden, hidden])
+
+    const editAction = useMemo(() => (
+        <DropdownActions.Item key="edit" onClick={onEdit}>
+            <Translate value="userpages.keyField.edit" />
+        </DropdownActions.Item>
+    ), [onEdit])
+
+    const deleteAction = useMemo(() => (
+        <DropdownActions.Item key="delete" onClick={onDelete} disabled={disableDelete}>
+            <Translate value="userpages.keyField.delete" />
+        </DropdownActions.Item>
+    ), [onDelete, disableDelete])
+
+    const inputActions = useMemo(() => ([
+        ...includeIf(!!hideValue, [revealAction]),
```

Comment:
is `!!` necessary since the falsey case is handled properly by ternary in `includeIf`

Reply:
Yea, need to give boolean to appease Flow